Global refactor of themes section (#2129)
PrestaShop/docs: themes/hummingbird/bootstrap-compatibility.md

---
title: Bootstrap compatibility
weight: 5
aliases:
  - /9/themes/reference/bootstrap-compatibility
---

# Bootstrap compatibility

## Why this matters

> Hummingbird uses [Bootstrap 5.3](https://getbootstrap.com/docs/5.3/getting-started/introduction/), but the PrestaShop module ecosystem was largely built on Bootstrap 4 (Classic theme). Bootstrap 5 introduced breaking changes — renamed classes, namespaced data attributes (`data-bs-*`), and the removal of jQuery as a dependency. Without intervention, modules that rely on BS4 patterns render incorrectly or lose interactive behavior (modals, tooltips, dropdowns) when used with Hummingbird.

Reviewer: ```suggestion
Hummingbird uses [Bootstrap 5.3](https://getbootstrap.com/docs/5.3/getting-started/introduction/), but the PrestaShop module ecosystem was largely built on Bootstrap 4 (Classic theme). Bootstrap 5 introduced breaking changes: renamed classes, namespaced data attributes (`data-bs-*`) and the removal of jQuery as a dependency. Without intervention, modules that rely on BS4 patterns render incorrectly or lose interactive behavior (modals, tooltips, dropdowns) when used with Hummingbird.
```

## The Bootstrap Compatibility Layer

> The [Bootstrap Compatibility Layer](https://github.com/PrestaShop/bootstrap-compatibility-layer) is a standalone library — it is **not part of Hummingbird**. It is a lightweight runtime shim that maps BS4 data attributes, class names, and jQuery plugin calls to their BS5 equivalents. Module developers who need it must include it in their own modules.

Reviewer: ```suggestion
The [Bootstrap Compatibility Layer](https://www.npmjs.com/package/bootstrap-compatibility-layer) is a standalone library, it is **not part of Hummingbird**. It is a lightweight runtime shim that maps BS4 data attributes, class names, and jQuery plugin calls to their BS5 equivalents. Module developers who need it must include it in their own modules.
```
I think that can be more worth to add the link to the NPM package instead of GitHub repo what do you think ?

{{% notice warning %}}

> The compatibility layer is a **temporary patch**, not a long-term solution. Migrating your module's markup to Bootstrap 5.3 is always the better approach — it removes a runtime dependency, reduces page weight, and avoids subtle rendering differences. Use the layer only as a stopgap while you plan a proper migration.

Reviewer: ```suggestion
The compatibility layer is a **temporary patch**, not a long-term solution. Migrating your module's markup to Bootstrap 5.3 is always the better approach. It removes a runtime dependency, reduces page weight, and avoids subtle rendering differences. Use the layer only as a stopgap while you plan a proper migration.
```

<script src="https://unpkg.com/bootstrap-compatibility-layer@1"></script>
```

### From a module hook

To load the compatibility layer from a module, use the `actionFrontControllerSetMedia` hook. Use `addCss()` and `addJs()` — these methods deduplicate by URL, so if multiple modules include the same asset it loads only once.

```php
public function hookActionFrontControllerSetMedia()
{
    $this->context->controller->addCss(
        'https://unpkg.com/bootstrap-compatibility-layer@1/dist/bootstrap-compatibility-layer.min.css'
    );
    $this->context->controller->addJs(
        'https://unpkg.com/bootstrap-compatibility-layer@1'
    );
}
```

## Usage

### As a package

```js
import BSCompatibilityLayer from 'bootstrap-compatibility-layer';
```

The layer initializes automatically on import.

### Via CDN

The library auto-initializes for attribute and class transformations. For jQuery-based Bootstrap calls, wait for DOM ready:

```html
<script src="https://unpkg.com/bootstrap-compatibility-layer@1"></script>
<script>
  document.addEventListener('DOMContentLoaded', function() {
    $('[data-toggle="popover"]').popover();
    $('[data-toggle="tooltip"]').tooltip();
  });
</script>

> ```

Reviewer: I'm really not sure about all of this. Everything is explained in the repository or on the package page, and it states right before that it's only temporary and should be avoided. I think we shouldn't elaborate further on how to use it, and even if we wanted to, it would be a subsection of "The Bootstrap Compatibility Layer," not a separate H2 heading.
